### PR TITLE
Setting user-agent header to python-gcsfs/<version>

### PR DESCRIPTION
**Pull request by vlasenkoalexey**
Replacement PR for #250

### CODE REVIEW

File: gcsfs/core.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 from requests.exceptions import RequestException, ProxyError
 from .utils import HttpError, is_retriable
+from ._version import get_versions
```
Comment (martindurant) [inline gcsfs/core.py]: `from . import __version__ as VERSION` and remove the `VERSION=` line below

Comment (vlasenkoalexey) [reply gcsfs/core.py]: I tried that, but it didn't work. As far as I understand it is causing circular dependency:

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-c3382239b67a> in <module>
----> 1 from gcsfs.core import GCSFileSystem, quote_plus, validate_response
      2 
      3 gcs = GCSFileSystem('alekseyv-scalableai-dev', token='/Users/alekseyv/.config/gcloud/application_default_credentials.json')
      4 gcs.ls('alekseyv-scalableai-dev')  # no error

~/vlasenkoalexey/gcsfs/gcsfs/__init__.py in <module>
----> 1 from .core import GCSFileSystem
      2 from .mapping import GCSMap
      3 from ._version import get_versions
      4 
      5 __version__ = get_versions()["version"]

~/vlasenkoalexey/gcsfs/gcsfs/core.py in <module>
     32 from .utils import HttpError, is_retriable
     33 #from ._version import get_versions
---> 34 from . import __version__ as VERSION
     35 
     36 logger = logging.getLogger(__name__)

ImportError: cannot import name '__version__'

Comment (martindurant) [reply gcsfs/core.py]: Ah indeed, the order in init needs changing too:
```diff
--- a/gcsfs/__init__.py
+++ b/gcsfs/__init__.py
@@ -1,8 +1,8 @@
-from .core import GCSFileSystem
-from .mapping import GCSMap
 from ._version import get_versions

 __version__ = get_versions()["version"]
 del get_versions
+from .core import GCSFileSystem
+from .mapping import GCSMap
```

Comment (vlasenkoalexey) [reply gcsfs/core.py]: this worked, thanks for suggestion

**Pull request merged**